### ISSUE TITLE
fetch resolves the promise but node process gets stuck

### ISSUE THREAD

**Issue by rzaharenkov** [bitinn/node-fetch#428]
I am faced with really weird issue. I do a fetch and it resolves the promise so I am getting page HTML. But nevertheless node.js process gets stuck. I guess it's something around node-fetch internals. It depends on the URL (I am running into this issue with 1 particular URL).

It's very easy to reproduce this issue.

```
// test.js

const fetch = require('node-fetch')
fetch('http://kjsubz.com').then(response => response.text()).then(console.log)
```

Then run it with `node test.js`

I've tried both v2.0 and v2.1 against different node.js versions (6.13.0, 8.9.1, 9.7.1)

**Comment by rzaharenkov** [comment]
OK, I think I know what is the reason of this issue. But not sure how to fix it properly.

In case of redirect it doesn't consume the response body (https://github.com/bitinn/node-fetch/blob/master/src/index.js#L130). And according to node.js documentation it must do it:

> If no 'response' handler is added, then the response will be entirely discarded. However, if a 'response' event handler is added, then the data from the response object must be consumed, either by calling response.read() whenever there is a 'readable' event, or by adding a 'data' handler, or by calling the .resume() method. Until the data is consumed, the 'end' event will not fire. Also, until the data is read it will consume memory that can eventually lead to a 'process out of memory' error.

I've tried to patch the code and it helped:

```
    // HTTP-redirect fetch step 15
    resolve(fetch(new Request(locationURL, requestOpts)));
+   res.on('data', () => {}) // do nothing but release handler
    return;
```

**Comment by bitinn** [comment]
Interesting, we will look into it.

(From my phone)

**Comment by TimothyGu** [comment]
@bitinn Two pull requests have been opened that fix this :) (#429 and #432).

**Comment by bitinn** [comment]
Yeah it’s not easy catching up via email.

Thx OP for going through the trouble of identifying the problem and making that PR though!

(From my phone)
